---
title: "Class12_RNAseq_with_DESeq2"
author: "Zixuan Zeng(A16142927)"
format: pdf
toc: true
---

```{r}
library(DESeq2, quietly=TRUE)
```

## Background

Today, we will analyze some RNAseq data from Himes et al. on the effects of a common steroid (dexmenthasone, AKA Dex) on airway smooth muscle cells (ASMs). 

For this analysis we need two main inputs

- `countData`: a table of **counts** per gene (in rows) across experiments (in columns).
- `colData`: **metadata** about the design of the experiments. The rows match the columns in `countData`.

## Data Import

```{r}
counts <- read.csv("airway_scaledcounts.csv", row.names=1)
metadata <- read.csv("airway_metadata.csv", row.names=1)
```

```{r}
head(counts)
metadata
```

>Q1. How many "genes" are in this dataset?

```{r}
nrow(counts)
```

>Q2. How many experiments (i.e columns in `counts` or rows in `metadata`) are in this dataset?

```{r}
nrow(metadata)
```

>Q3. How many "control" experiments are there in this dataset?

```{r}
sum(metadata$dex == "control")
```

1. Extract the "control" columns from `counts`.
2. Calculate the mean value for each genein these columns.

3-4. Do the same for the "treated" columns.

5. compared the means between `control` and `treated` for each gene.

```{r}
control.inds <- metadata$dex == "control"
control.counts <- counts[,control.inds]
control.means <- rowMeans(control.counts)
```

```{r}
treated.inds <- metadata$dex == "treated"
treated.counts <- counts[,treated.inds]
treated.means <- rowMeans(treated.counts)
```

Fpr ease pf book-keeping we can store these together in one data frame called `meancounts`

```{r}
meancounts <- data.frame(control.means,treated.means)
head(meancounts)
```


```{r}
library(ggplot2)
ggplot(meancounts) + aes(control.means,treated.means) + geom_point() + scale_x_log10() + scale_y_log10()
plot(meancounts,log="xy")
```


We use "fold-change" as a way to compare.

```{r}
#treated/control
head(log2(treated.means/control.means))
```

```{r}
meancounts$log2fc <- log2(meancounts$treated.means / meancounts$control.means)
head(meancounts)
```

A common "rule-of-thumb" threshold for calling someing "upregulated" is a log2-fold-change of +2 or more, and vice versa.

```{r}
zero.inds <- which(meancounts[,1:2] == 0,arr.ind=TRUE)[,1]
mygenes <- meancounts[-zero.inds, ]
```


>Q. How many genes are upregulated at the +2 threshold?

```{r}
sum(mygenes$log2fc >= 2, na.rm=TRUE)
```

>Q. How many genes are downregulated at the -2 threshold?

```{r}
sum(mygenes$log2fc <= -2, na.rm=TRUE)
```


## DeSeq analysis

Let's do this with DESeq2 and put some stats behind these numbers. DESeq wants 3 things for analysis: countData, colData, and design.

```{r}
dds <- DESeqDataSetFromMatrix(countData = counts, colData = metadata, design = ~ dex)
```

The main function in the DESeq package to run analysis is called `DESeq()`.

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
head(res)
```

## Volcano Plot

```{r}
plot(res$log2FoldChange,-log(res$padj))
abline(v=c(-2,2),col="red")
abline(h=-log(0.05),col="red")
```

## Save our results

```{r}
write.csv(res,file="myreults.csv")
```


## A nicer ggplot volcano plot

```{r}
mycols <- rep("gray",nrow(res))
mycols[(res$log2FoldChange) > 2 ] <- "red"
mycols[(res$log2FoldChange) < -2 ] <- "blue"
mycols[res$padj >= 0.05] <- "gray"
ggplot(res) + aes(log2FoldChange,-log(padj)) + geom_point(col=mycols) +geom_vline(xintercept=c(-2,2),col="black") + geom_hline(yintercept=-log(0.05),col="black") + theme_classic()
```

## Add annotation data

We need to add gene symbols, gene names and other database ids to make my results useful for further analysis.

```{r}
library("AnnotationDbi")
library("org.Hs.eg.db")
```



Let's see what databse id formats we can translate between 
```{r}
columns(org.Hs.eg.db)
```


```{r}
res$symbol <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",   # The format of our genenames
                     column="SYMBOL",     # The new format we want to add
                     multiVals="first")
```

```{r}
head(res$symbol)
```

Add `GENENAME` and `ENTREZID`

```{r}
res$genename <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",   # The format of our genenames
                     column="GENENAME",   # The new format we want to add
)
head(res$genename)
```

```{r}
res$entrezid <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",   # The format of our genenames
                     column="ENTREZID",   # The new format we want to add
)
head(res$entrezid)
```

## Save my annotated results

```{r}
write.csv(res,file = "myresults_annotated.csv")
```


## Pathway analysis

We will use the **gage** function from bioconductor.

What **gage** wants as input is a named vector of importance (a vector with labeled fold-changes)

```{r}
library(pathview)
library(gage)
library(gageData)

data(kegg.sets.hs)

# Examine the first 2 pathways in this kegg set for humans
head(kegg.sets.hs, 2)
```

```{r}
foldchanges <- res$log2FoldChange
names(foldchanges) <- res$entrezid
head(foldchanges)
```

```{r}
data(kegg.sets.hs)
keggres = gage(foldchanges,gsets=kegg.sets.hs)
```

```{r}
head( keggres$less, 5 )
```


Let's look at just one of these

```{r,message=F}
pathview(gene.data = foldchanges,pathway.id = "hsa05310")
```

Insert figure for this pathway

![Asthma pathway from KEGG with my differentially expressed genes highlighted](hsa05310.pathview.png)